docs/collecting-data/collecting-from-own-applications/mobile-trackers/migration-guides/migration-guide-from-version-5-x-to-6-0/index.md

```diff
@@ -0,0 +1,82 @@
+
+There's no change to the behaviour of the variable `entities`, so you could still replace them all using `event.entities = newListEntities`.
+
+## Non-optional tracker on iOS
+
+In the v5.x of the iOS tracker, `createTracker` returned an optional `TrackerController?`. This was an oversight. In v6, the iOS tracker again returns a non-optional `TrackerController`.
+
+## Concurrency on iOS
+
+The return type of the `Tracker.track()` method has changed from `UUID?` to `UUID`. Previously, `nil` could be returned if the tracker was paused, or if the event was filtered out and not actually tracked. From v6 onwards it will always return a UUID. If the event is tracked, this will be the `eventId`.
+
+This change was necessary as part of the comprehensive refactoring of the tracker thread model. The iOS tracker now uses a global dispatch queue. This single queue makes the tracker much safer. Almost all actions are now performed concurrently. Network requests in Emitter are still asynchronous.
+
+## Changes to the `EventStore` interface
+
+A new method, `removeOldEvents()` has been added to the `EventStore` protocol on both trackers. This method is used in the new feature that deletes events from storage if they get too old (by default, 30 days). Also, if too many events collect in the `EventStore`, the older ones will be deleted (default 1000 events).
+
+For the Android tracker, we have updated the `EventStore` interface to remove the optional types.
+
+## Non-breaking changes and bug fixes
+
+### New events
+
+The screen engagement ADD LINK feature adds new events for both iOS and Android. For iOS only, there are new events (and a new demo app) for visionOS ADD LINK. For Android, the `PageView` event has been restored, after accidental deprecation in v5.
+
+### Cross-device tracking
+
+Decorate URIs in both trackers using the new `CrossDeviceParameterConfiguration`. ADD LINK 
+
+### Emitter and network connection behaviour
+
+The Android tracker default emit timeout has been increased to 30 seconds, from 5 seconds. This setting is configured ADD LINK using the `NetworkConfiguration`. The `timeout` option has been added to the iOS tracker.
+
+In both trackers, we have officially set the default batch size to 1 (`BufferOption.Single`), i.e. the events are sent as soon as they are tracked. This is consistent with the web tracker, and is a good default to use in client side apps to prevent some events hanging in the event store when the app is quit/uninstalled earlier than the event store is cleared. In theory, the previous behaviour was to send batches of 10 (`BufferOption.SmallGroup`), but due to a bug, they were always sent individually. We've also fixed a bug in which the `bufferOption` in `EmitterConfiguration` was not being used correctly, making it impossible to batch events. This is now possible.
```

Comment:
I think this is a breaking change for two reasons:
- `BufferOption.defaultGroup` was renamed to `BufferOption.smallGroup`
- If the user specifically configured `defaultGroup` or `largeGroup` previously, the behaviour will change for them. While before events were sent right after tracked, now the tracker will wait until 10 or 25 events accumulate in the event store. You can link to the new updated section in another PR.